#2 mind builder

MMCQ33 opened the pull request:
Broken UI, and needs added save button. But minds can be build and edited.

MMCQ33 committed:
First version of mind building

Added ant class for all ants. Added basic minds which can be changed with settings. Modified gathering so it isnt a unity script anymore.

MMCQ33 committed:
Merge remote-tracking branch 'origin/Develop' into #2-MindBuilder

MMCQ33 committed:
Formula version 1

Made the first version of the formula that defines the ant speed when picking up resources.

MMCQ33 committed:
Fixed merge

fixed merge with map prefab changes

MMCQ33 committed:
First MindBuilder UI

Added icons and layers of the mind builder, readded the pause menu because it was gone.

MMCQ33 committed:
Merge branch 'Develop' into #2-MindBuilder

MMCQ33 committed:
fixed merge with develop

Made mind builder work with develop code.

MMCQ33 committed:
Made general Mind object and works with the ANTS

Made basic mind structure and reformatted the ant, mind and the groups. Ants made from the menu now has the same minds as their groups and mind groups.

MMCQ33 committed:
Prettified

prettified code.

MMCQ33 committed:
Fixed bug

Ants would consider their own position their storage.

jetenergy committed:
Removed junk scenes, changed build settings, changed all the UI

jetenergy committed:
Removed unused nav mesh, fixed all broken UI

MMCQ33 committed:
Changed dynamic minds

Made minds dynamic. Adding minds is possible, update old minds, Fixed bugs.

MMCQ33 committed:
Merge branch '#2-MindBuilder' into #2-Mindbuilder-UI

MMCQ33 committed:
Revert "Removed unused nav mesh, fixed all broken UI"

This reverts commit 6832bb4899fb4721ef6016346a75c35caa3b75c5.

MMCQ33 committed:
Added UI

Added UI. Made it Variable with enums. Saves minds correctly. Close Button saves minds. Ants take the correct values.

MMCQ33 committed:
Merge remote-tracking branch 'origin/Develop' into #2-MindBuilder

MMCQ33 committed:
Fixed merge broken

Fixed stuff that was broke through merge

MMCQ33 committed:
Fixed Map and ui

Fixed map and fixed broken UI in the map.

ErikLouwerse commented on Hivemind/Assets/Scripts/Ant.cs:
```diff
@@ -0,0 +1,121 @@
+            }
+        }
+
+        double likeliest= 0;
```
Spacing between =
Recommend some automated cleanup on all changed document

ErikLouwerse commented on Hivemind/Assets/Scripts/MindGroup.cs:
```diff
@@ -15,9 +15,25 @@ public class MindGroup
+    
+    public int mindPoints { get; set; }
+
+    public MindGroup(GameObject UiObject, IMind mind = null)
```
mind in constructor is not used

ErikLouwerse commented on Hivemind/Assets/Scripts/MindGroup.cs:
```diff
@@ -15,9 +15,25 @@ public class MindGroup
-    public MindGroup(GameObject UiObject)
+    public IMind Mind { get; private set; } = new Gathering(ResourceType.Unknown, 1, Gathering.Direction.None);
+
+    public List<IMind> minds { get; private set; }
```
Public variables should start with a capital letter

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/CombatFight.cs:
```diff
@@ -0,0 +1,9 @@
+﻿using System;
```
Unused using, recommend automated cleanup for all files

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/CombatMind.cs:
```diff
@@ -0,0 +1,94 @@
+﻿
+public class CombatMind : IMind
+{
+    private float minEstimetedDifference;
```
Estimated

ErikLouwerse replied:
Use visual studios Rename so you don't have to manually change everything

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/CombatMind.cs:
```diff
@@ -0,0 +1,94 @@
+                 new CombatFight().Execute(ant);
+            }
+
+            float healthPercantageDifference = ((float)ant.health / (float)ant.closestEnemy.health);
```
Percentage

ErikLouwerse replied:
Also to get the percentage you should do * 100

ErikLouwerse replied:
And also -100, recommend removing the percentage word from this variable and the one below

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/CombatMind.cs:
```diff
@@ -0,0 +1,94 @@
+            {
+                 new CombatFight().Execute(ant);
+            }
+             new CombatFlee().Execute(ant);
```
Shouldn't this be in an else?

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -29,105 +30,116 @@ public enum Direction
-    private float baseSpeed;
-    private List<GameObject> carryingObjects;
+
+    public List<GameObject> carryingObjects;
```
Variable is only used in this class, should revert to private

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -29,105 +30,116 @@ public enum Direction
+        IsScout = isScout;
+    }
+
+    private ResourceNode findResource(ResourceType PrefferedResource)
```
Why give the PrefferedResource when you already have acces to it in the class?

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -190,26 +209,27 @@ private void Update()
                 }
                 else
                 {
-                    storage = GameWorld.GetStorage();
```
What if storage cannot be found?

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/CombatMindUI.cs:
```diff
@@ -0,0 +1,30 @@
+
+public class CombatMindUI : MonoBehaviour, IMindUI
+{
+    public GameObject Ui_IconObj { get; set ; }
```
Remove Useless getset

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/IMindUI.cs:
```diff
@@ -0,0 +1,15 @@
+
+interface IMindUI
+{
+    GameObject Ui_IconObj { get; set; }
```
remove getset

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/MindBuilderScript.cs:
```diff
@@ -0,0 +1,112 @@
+    public MindGroup mindGroup;
+
+    // Start is called before the first frame update
+    void Start()
```
Remove unused function

ErikLouwerse commented on Hivemind/Assets/Scripts/UI/ResourceMindUI.cs:
```diff
@@ -0,0 +1,36 @@
+
+public class ResourceMindUI : MonoBehaviour, IMindUI
+{
+    public GameObject Ui_IconObj { get; set; }
```
getset

ErikLouwerse commented on Hivemind/Assets/Scripts/Ui/UiController.cs:
```diff
@@ -24,6 +24,8 @@ public class UiController : MonoBehaviour, IInitializePotentialDragHandler, IDra
     [SerializeField]
     private GameObject mindBuilderPanel;
 
+    private MindGroup CurrentOpenMindGroup;
```
private field should not start with capital letter

MMCQ33 committed:
Formatting and small fixes

Formatting changes. Removed unused code, Edited code towards standards. Made UI more scaleable.

MMCQ33 committed:
final UI changes

UI looks good when scaleable

MMCQ33 committed:
Used resharper and fixed scouting

Used resharper on all edited files and made ants not idle when all resources are gone

MMCQ33 committed:
Added Save Button

Added Save button that saves the mind
